pilot/pkg/bootstrap/server.go

```diff
@@ -162,14 +163,16 @@ type Server struct {
 	mixerSAN         []string
 	kubeClient       kubernetes.Interface
 	startFuncs       []startFunc
-	clusterStore     *clusterregistry.ClusterStore
 	httpServer       *http.Server
 	grpcServer       *grpc.Server
 	secureGRPCServer *grpc.Server
 	discoveryService *envoy.DiscoveryService
 	istioConfigStore model.IstioConfigStore
 	mux              *http.ServeMux
-	kubeRegistry     *kube.Controller
+	kubeRegistry     map[string]*kube.Controller
+	ResyncPeriod     time.Duration
+	WatchedNamespace string //TODO - is there a better place to store these?
```

Comment:
I really feel strongly that the functionality should remain in the original directory so that stuff related to cluster management can be encapsulated there. Things such as those variables, and per-cluster stop channel, as well as status of each cluster that can be exposed to some debug command, etc.